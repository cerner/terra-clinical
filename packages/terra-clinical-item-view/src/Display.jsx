import React, { PropTypes } from 'react';
import classNames from 'classnames';
import 'terra-base/lib/baseStyles';
import './Display.scss';

const textStyles = [
  'secondary',
  'attention',
  'strikeThrough',
];

const propTypes = {
  /**
   * The text to be displayed for the comment.
   */
  text: PropTypes.string,
  /**
   * The visual styles to be applied to the display element.
   */
  textStyle: PropTypes.oneOf(textStyles),
  /**
   * Whether or not the text should be truncated.
   */
  isTruncated: PropTypes.bool,
  /**
   * The icon react element to be displayed next to the display text.
   */
  icon: PropTypes.element,
};

const defaultProps = {
  text: '',
  textStyle: undefined,
  isTruncated: false,
  icon: undefined,
};

const Display = ({
    text,
    textStyle,
    isTruncated,
    icon,
    ...customProps
  }) => {
  const displayClassNames = classNames([
    'terraClinical-ItemView-display',
    customProps.className,
  ]);

  const textClassNames = classNames([
    'terraClinical-ItemView-text',
    { 'terraClinical-ItemView-text--isTruncated': isTruncated },
    { [`terraClinical-ItemView-text--${textStyle}`]: textStyle },
  ]);

  let displayIcon;
  if (icon) {
    displayIcon = <div className="terraClinical-ItemView-text--inlineIcon">{icon}</div>;
  }

  return (
    <div {...customProps} className={displayClassNames}>
      {displayIcon}
      <div className={textClassNames}>{text}</div>
    </div>
  );
};

Display.propTypes = propTypes;
Display.defaultProps = defaultProps;

export default Display;
