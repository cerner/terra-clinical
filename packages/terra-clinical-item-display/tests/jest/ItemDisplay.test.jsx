import React from 'react';
import Display from '../../src/ItemDisplay';

// Snapshot Tests
it('should render a default component', () => {
  const itemDisplay = shallow(<Display />);
  expect(itemDisplay).toMatchSnapshot();
});

it('should render with text', () => {
  const itemDisplay = shallow(<Display text="Display" />);
  expect(itemDisplay).toMatchSnapshot();
});

it('should render with a graphic', () => {
  const graphic = <img alt="Graphic" />;
  const itemDisplay = shallow(<Display icon={graphic} />);
  expect(itemDisplay).toMatchSnapshot();
});

it('should render with text and graphic center aligned', () => {
  const graphic = <img alt="Graphic" />;
  const itemDisplay = shallow(<Display text="Display" icon={graphic} iconAlignment="center" />);
  expect(itemDisplay).toMatchSnapshot();
});

it('should render with text and graphic top aligned', () => {
  const graphic = <img alt="Graphic" />;
  const itemDisplay = shallow(<Display text="Display" icon={graphic} iconAlignment="top" />);
  expect(itemDisplay).toMatchSnapshot();
});

it('should render with text and graphic inline aligned', () => {
  const graphic = <img alt="Graphic" />;
  const itemDisplay = shallow(<Display text="Display" icon={graphic} iconAlignment="inline" />);
  expect(itemDisplay).toMatchSnapshot();
});

it('should render disabled display with text and graphic', () => {
  const graphic = <img alt="Graphic" />;
  const itemDisplay = shallow(<Display text="Display" icon={graphic} isDisabled />);
  expect(itemDisplay).toMatchSnapshot();
});

it('should render a truncated display', () => {
  const itemDisplay = shallow(<Display text="display1display1display1display1display1display1display1display1" isTruncated />);
  expect(itemDisplay).toMatchSnapshot();
});

it('should render a disabled display', () => {
  const itemDisplay = shallow(<Display text="Display" isDisabled />);
  expect(itemDisplay).toMatchSnapshot();
});

it('should render a display with the primary textStyle on text', () => {
  const itemDisplay = shallow(<Display text="Display" textStyle="primary" />);
  expect(itemDisplay).toMatchSnapshot();
});

it('should render a display with the attention textStyle on text', () => {
  const itemDisplay = shallow(<Display text="Display" textStyle="attention" />);
  expect(itemDisplay).toMatchSnapshot();
});

it('should render a display with the secondary textStyle on text', () => {
  const itemDisplay = shallow(<Display text="Display" textStyle="secondary" />);
  expect(itemDisplay).toMatchSnapshot();
});

it('should render a display with the strikeThrough textStyle on text', () => {
  const itemDisplay = shallow(<Display text="Display" textStyle="strikeThrough" />);
  expect(itemDisplay).toMatchSnapshot();
});

it('should render a display with the strong textStyle on text', () => {
  const itemDisplay = shallow(<Display text="Display" textStyle="strong" />);
  expect(itemDisplay).toMatchSnapshot();
});
