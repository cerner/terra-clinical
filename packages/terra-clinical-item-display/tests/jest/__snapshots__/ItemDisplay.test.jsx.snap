exports[`test should render a default component 1`] = `
<div
  className="item-display">
  <div
    className="text" />
</div>
`;

exports[`test should render a display with the attention textStyle on text 1`] = `
<div
  className="item-display">
  <div
    className="text attention">
    Display
  </div>
</div>
`;

exports[`test should render a display with the secondary textStyle on text 1`] = `
<div
  className="item-display">
  <div
    className="text secondary">
    Display
  </div>
</div>
`;

exports[`test should render a display with the strikeThrough textStyle on text 1`] = `
<div
  className="item-display">
  <div
    className="text strikeThrough">
    Display
  </div>
</div>
`;

exports[`test should render a truncated display 1`] = `
<div
  className="item-display">
  <div
    className="text is-truncated">
    display1display1display1display1display1display1display1display1
  </div>
</div>
`;

exports[`test should render with a graphic 1`] = `
<div
  className="item-display">
  <div
    className="inline-icon">
    <img
      alt="Graphic" />
  </div>
  <div
    className="text" />
</div>
`;

exports[`test should render with text 1`] = `
<div
  className="item-display">
  <div
    className="text">
    Display
  </div>
</div>
`;

exports[`test should render with text and graphic 1`] = `
<div
  className="item-display">
  <div
    className="inline-icon">
    <img
      alt="Graphic" />
  </div>
  <div
    className="text">
    Display
  </div>
</div>
`;
